### ISSUE TITLE
Support CLP(B)

### ISSUE THREAD

**Issue by jgarte** [trealla-prolog/trealla#564]
Hi, 

Is supporting CLP(Z) just a matter of vendoring in Markus Triska's [clpz.pl](https://github.com/triska/clpz/blob/master/clpz.pl) file?

What would be required at the lower level C code in order to support it.

Any orientative explanation of what would be required would be much appreciated so I can then dig deeper into understanding the requirements.

all best,

jgart

**Comment by infradig** [comment]
Hi, I'm not sure what you mean, as library/clpz.pl is already included in
the Trealla release.

**Comment by jgarte** [comment]
Oops, for some reason I thought it was not included yet. I must have missed that file when looking.

Is there any further work to do on `clpz.pl` in trealla or is it at feature parity with scryer's support for CLP(Z)?

**Comment by infradig** [comment]
As far as I can tell it works as expected. CLP(B) however is still missing
as it doesn't work properly yet.

**Comment by jgarte** [comment]
Thanks for the info! That's great that CLP(Z) is working as expected on trealla. I'm looking forward to using the library more as I am currently learning [declarative integer arithmetic](https://www.metalevel.at/prolog/clpz).

What's left to do on CLP(B) or what is the current difficulty with it?

**Comment by infradig** [comment]
Unknown problem that I can't seem to get to the bottom of. I'll return to
it soon. CLP(B) & Tabling are the outstanding problems.

**Comment by jgarte** [comment]
I've changed this issue name to "Support CLP(B)" since I saw that there is no ticket open for it currently.

**Comment by jgarte** [comment]
What is Tabling? Can you point me to a reference on tabling so that I can orient myself to it. Thanks!

**Comment by jgarte** [comment]
You mean this?

https://en.wikipedia.org/wiki/Tabled_logic_programming

**Comment by infradig** [comment]
Yes, the Scryer code for tabling should be portable but doesn't work.

**Comment by jgarte** [comment]
Oh, ok 

https://www.scryer.pl/tabling

**Comment by jgarte** [comment]
Hi,

Shouldn't the clpb module be included at this path?

https://github.com/trealla-prolog/trealla/tree/main/library

prolog module source: https://github.com/mthom/scryer-prolog/blob/master/src/lib/clpb.pl

**Comment by infradig** [comment]
It's not functional so it's not included.
